haproxy/Dockerfile

```diff
@@ -0,0 +1,78 @@
+FROM ghcr.io/cybozu/ubuntu:20.04 AS build
+
+ENV HAPROXY_BRANCH 2.6
+ENV HAPROXY_MINOR 2.6.10
+ENV HAPROXY_SHA256 e71b2cd9ca1043345f083a5225078ccf824dced2b5779d86f11fa4e88f451773
+ENV HAPROXY_SRC_URL http://www.haproxy.org/download
+ENV DEBIAN_FRONTEND noninteractive
+
+# if you use ubuntu:22.04, you must use "libssl3=3.0.2-0ubuntu1.9". but HA-Proxy 2.6 does NOT adapt it.
+# We choose "ubuntu:20.04" and "libssl1.1" to get a stable operation.
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    procps \
+    zlib1g \
+    libssl1.1 \
+    "libpcre2-*" \
+    liblua5.3-0 \
+    libatomic1 \
+    tar \
+    curl \
+    socat \
+    gcc \
+    make \
+    libc6-dev \
+    libssl-dev \
```

Comment:
It seems to be simple by using "ghcr.io/cybozu/ubuntu-dev:20.04". it is good idea, so, I confirmed Github "https://github.com/cybozu/ubuntu-base" and then I have a concern for this change.

The concern is that using this image hides the modules required by HA-PROXY. This makes maintenance difficult when changes are made to the upstream HA-PROXY.

We hope to ensure maintainability by using a minimal Ubuntu image and describe each module in Dockerfile.

How do you think about this ?

Reply:
Since the HAProxy documentation describes how to build, how about writing that documentation in the README?
https://github.com/haproxy/haproxy/blob/v2.6-dev10/INSTALL

Reply:
HA-Proxy has been in use for over 20 years and has been in use since before Docker containers. This installation document is for direct installation on Server OS.

I have been unable to locate documentation that addresses the considerations for containerizing HA-Proxy. However, I believe it is described in the HA-Proxy Dockerfile.

To avoid known problems, I believe it is clever to develop on the basis of the HA-Proxy Dockerfile. I believe that the knowledge of HA-Proxy for containers will be accumulated in that Dockerfile in the future.